### DISCUSSION TITLE
go2rtc/ffmpeg cpu usage monitoring in web interface

### DISCUSSION THREAD

**Discussion by tve** [blakeblackshear/frigate#15230] [Ideas]
Running 0.15.0-beta2 I had go2rtc transcode an mjpeg stream to h264 using ffmpeg without hardware acceleration and it was using a steady 30% of cpu. I saw 'ffmpeg' in top but since there are umpteen ffmpeg processes running I couldn't tell what it was. I looked in the frigate system metrics and it didn't show either! (But it did show "CPU:30%" in the footer.) Unless I'm missing something, it would be nice to add a section that shows go2rtc cpu usage, including child processes.

**Comment by hawkeye217** [comment]
Under the General tab in System Metrics, you should have a "Process CPU Usage" column, including a go2rtc graph.
![Screenshot 2024-11-27 at 10 11 32 PM](https://github.com/user-attachments/assets/566c6236-d9a3-4dc9-8527-ce9c041a7e16)

**Reply by tve** [reply]
Yup, you are correct! However, this did not show the 30% cpu that were consumed by the ffmpeg subprocess of go2rtc which was transcoding without hardware acceleration.

**Reply by hawkeye217** [reply]
Right, because go2rtc just runs ffmpeg when you do transcoding with `ffmpeg:rtsp://.../#video=h264` in your go2rtc config.

**Reply by NickM-27** [reply]
In order for us to do this we'd need a more specific tool that tells us what ffmpeg processes belong to go2rtc, then we could show them on the camera specific page, not sure how possible that will be
